src/main/java/com/epam/reportportal/extension/azure/AzureExtension.java

```diff
@@ -231,19 +233,40 @@ private void removeListeners() {
 
 	@Override
 	// Never called method. Connection is tested via the command.
-	public boolean testConnection(Integration system) {
+	public boolean testConnection(Integration integration) {
 		return false;
 	}
 
 	@Override
-	// TODO: Implement in the future
-	public Optional<Ticket> getTicket(String id, Integration system) {
-		return Optional.empty();
+	public Optional<Ticket> getTicket(String id, Integration integration) {
+		Map<String, Object> params = integration.getParams().getParams();
+		String organizationUrl = params.get("url").toString();
+		String projectName = params.get("project").toString();
+		String personalAccessToken = params.get("oauthAccessKey").toString();
+
+		ApiClient defaultClient = getConfiguredApiClient(personalAccessToken);
+		String organizationName = organizationUrl.replace(defaultClient.getBasePath(), "");
+
+		WorkItemsApi workItemsApi = new WorkItemsApi(defaultClient);
+		try {
+			WorkItem workItem = workItemsApi
+					.workItemsGetWorkItem(organizationName, Integer.valueOf(id), projectName, API_VERSION,
+							null, null, null);
+			Ticket ticket = new Ticket();
+			ticket.setId(workItem.getId().toString());
+			ticket.setTicketUrl(workItem.getUrl());
+			ticket.setStatus(workItem.getFields().get("System.State").toString());
+			ticket.setSummary(workItem.getFields().get("System.Title").toString());
+			return Optional.of(ticket);
+		} catch (ApiException e) {
+			LOGGER.error("Unable to load ticket: " + e.getMessage(), e);
+			return Optional.empty();
```

Comment:
what happens on UI when you return empty? shouldn't we throw an exception instead?

Reply:
We shouldn't throw an exception, because the Report Portal's `getTicket()` method, which invoke plugin's `getTicket()` method, throws an exception by itself if optional is empty.
![image](https://user-images.githubusercontent.com/68337846/129566367-0848ccb4-5be4-464f-9f85-80b33ae776f1.png)